src/utils/uintarray-encoding.js

```diff
@@ -162,16 +300,24 @@ function encodingCharToNumber(x: string): number {
   }
 }
 
-export function stringToUintArray(s: string): number[] {
-  const array = [];
-  let val = 0;
-  for (let i = 0; i < s.length; i++) {
-    const x = encodingCharToNumber(s[i]);
-    val = (val << 5) + (x & 0b11111);
-    if ((x & 0b100000) === 0) {
-      array.push(val);
-      val = 0;
-    }
+function decodeUint(
+  s: string,
+  start: number
+): {| value: number, hasLeadingZero: boolean, nextI: number |} {
+  let i = start;
+  let bits = bitsFromEncodingDigit(s[i]);
+  let continuationBit = bits & 0b100000;
+  let valueBits = bits & 0b011111;
+  const hasLeadingZero = continuationBit !== 0 && valueBits === 0;
+  let value = valueBits;
+  let nextI = i + 1;
```

Comment:
nit: why don't you increment `i` here and line 320 and use `i` directly in the `while` loop and the return value?

Reply:
Heh, yeah, not sure what I was thinking.